coalib/io/FileFactory.py

```diff
@@ -0,0 +1,115 @@
+import linecache
+import os
+
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('_filename', '_stamp')
+class FileFactory:
+    """
+    The FileFactory is a wrapper object to replace the actual file contents
+    from the file-dict. The object contains the following information about
+    the file:
+
+        * The filename (absolute file path).
+        * The parent directory the file is located in.
+        * The modification timestamp of the file.
+        * The file in the form of a string.
+        * The file in the form of a list.
+        * The file in binary format.
+
+    The equality of the object is checked just based on its filename and
+    timestamp.
+
+    To initialize a FileFactory object for a file:
+
+    >>> fp = FileFactory("./tests/io/FileFactoryTestFiles/test.txt")
+
+    Make the object into an iterable when required
+
+    >>> for line in fp:
+    ...     print(line)
+    This is a test file.
+    <BLANKLINE>
+
+    Retrieve a single line:
+
+    >>> fp.line(1)
+    'This is a test file.\\n'
+
+    Get all the lines in the file as a list:
+
+    >>> fp.lines
+    ('This is a test file.\\n',)
+    """
+
+    def __init__(self, filename):
+        """
+        :param filename:
+            The absolute filepath.
+        """
+        self._filename = os.path.abspath(filename)
+        self._parent = os.path.abspath(os.path.dirname(self._filename))
```

Comment:
Do we really need this variable as data member ?
This is easily deducible whenever required.

Reply:
We might need it in phase 2 for the ignore directory functionality though I'm still not sure as I need to have a lot of discussion about its design with Makman so I was thinking about letting it stay there and if we don't need it in phase 2 then I will remove it.